composite/market-closure/src/checks/index.ts

```diff
@@ -0,0 +1,44 @@
+import { scheduleExecute } from './schedule'
+import { thExecute } from './tradinghours'
+import { Schedule } from 'market-closure'
+
+export type CheckExecute = (symbol: string, schedule: Schedule) => Promise<boolean>
+
+export type ExternalCheck = (symbol: string) => Promise<boolean>
+export type ScheduleCheck = (schedule: Schedule) => boolean
+
+export type CheckOptions = { type?: Check }
+export enum Check {
+  Schedule = 'schedule',
+  TradingHours = 'tradinghours',
+}
```

Comment:
Some of those types are unnecessary:

```suggestion
// We check for something and get yes/no answer
export type Check = () => Promise<boolean>

export enum CheckProvider {
  Schedule = 'schedule',
  TradingHours = 'tradinghours',
}
```